### DISCUSSION TITLE
fastdds v2.4 build error

### DISCUSSION THREAD

**Discussion by gitguoqun** [eProsima/Fast-DDS#5270] [General]
fastddsgen:v2.0.2     foonathan_memory_vendor:version: v1.1.0  fastcdr: version: v1.0.22   fastrtps:version: v2.4.0    
when i build fastdds example/C++/HelloWorldExample   a error conmes out:
In file included from /home/dfrd/work/3rdLib/demo/fastdds/3/HelloWorldPubSubTypes.cxx:26:
/home/dfrd/work/3rdLib/demo/fastdds/3/HelloWorldPubSubTypes.h:32:2: error: #error Generated HelloWorld is not compatible with current installed Fast DDS. Please, regenerate it with fastddsgen.
   32 | #error Generated HelloWorld is not compatible with current installed Fast DDS. Please, regenerate it with fastddsgen.
      |  ^~~~~
make[2]: *** [CMakeFiles/HelloWorldExample.dir/build.make:90: CMakeFiles/HelloWorldExample.dir/HelloWorldPubSubTypes.cxx.o] Error 1

**Comment by JesusPoderoso** [comment]
Hi @gitguoqun, thanks for using Fast DDS.

Actually, Fast DDS v2.4.0 is EOL. Refer to [the release support](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md) document for related information. Do you have any other Fast DDS version sourced in the environment?
Can you check if the issue persists with the currently supported versions (2.6.x, 2.10.x, 2.14.x and 3.0.x)?

**Comment by gitguoqun** [comment]
Hi @JesusPoderoso  Thanks for your replay.
Yes i can the 2.10.4 successfully.

By the way there is an other question about the fastdds.  i can not recive the fast dds topic with my wsl ubuntu. Do you known the reaseon?

**Reply by JesusPoderoso** [reply]
I would need more information about the deployment to determine why your connection fails @gitguoqun
